Projects/MouseControl/mouse_control_movement.py

```diff
@@ -0,0 +1,154 @@
+    x_axis, y_axis = struct.unpack("bb", buf[1:])
+
+    if debug is True:
+        print("Left but: {}, Middle but: {}, Right but: {}, x pos: {}, y pos: {}\n".format(left_button, middle_button, right_button, x_axis, y_axis))
+
+    return (left_button, middle_button, right_button, x_axis, y_axis)
+
+
+def Main():
+
+    print("   _____       _____ _  _____         ____  ")
+    print("  / ____|     |  __ (_)/ ____|       |___ \ ")
+    print(" | |  __  ___ | |__) || |  __  ___     __) |")
+    print(" | | |_ |/ _ \|  ___/ | | |_ |/ _ \   |__ < ")
+    print(" | |__| | (_) | |   | | |__| | (_) |  ___) |")
+    print("  \_____|\___/|_|   |_|\_____|\___/  |____/ ")
+    print("                                            ")
+
+    print("For moving the robot around using the mouse buttons press 1 and enter.")
+    print("For moving the robot around using the movements of the mouse press 2 and enter.")
+
+    try:
+        choice = int(input("choice (1/2) = "))
+    except ValueError:
+        print("Invalid number read")
+        sys.exit(1)
+
+    if not (choice == 1 or choice == 2):
+        print("Invalid number entered")
+        sys.exit(1)
+
+    print("\nWith this script you can control your GoPiGo3 robot with nothing but a mouse.")
+    if choice == 1:
+        print("1. Left + Right buttons of the mouse - move the GoPiGo3 forward")
+        print("2. Left button of the mouse - move the GoPiGo3 to the left")
+        print("3. Right button of the mouse - move the GoPiGo3 to the right")
+        print("4. Middle button of the mouse - move the GoPiGo3 backward")
+    else:
+        print("1. Move the mouse forward - for moving the GoPiGo3 forward")
+        print("2. Move the mouse backward - for moving the GoPiGo3 backward")
+        print("3. Move the mouse to the left - for rotating the GoPiGo3 to the left")
+        print("4. Move the mouse to the right - for rotating the GoPiGo3 to the right")
+
+    # Wait for an input to start
+    input("Press Enter to start")
+
+    try:
+        robot = EasyGoPiGo3()
+
+    except IOError:
+        print("GoPiGo3 not detected")
+        sys.exit(1)
+
+    except gopigo3.FirmwareVersionError:
+        print("Please update your GoPiGo3 firmware")
+        sys.exit(1)
+
+    except Exception:
+        print("Something went wrong")
+        sys.exit(1)
+
+    print("\nIn order to stop the script, press CTRL-C and move your mouse a little bit")
+    print('This script has blocking-methods and moving the mouse will get the script "moving" ')
+
+    with open("/dev/input/mice", "rb") as mouse_input:
+
+        left_button = 0
+        middle_button = 0
+        right_button = 0
+        x_axis = 0
+        y_axis = 0
+
+        while signal_not_called:
+
+            (left_button, middle_button, right_button, x_axis, y_axis) = getMouseValues(mouse_input)
+
+            if choice == 1:
+
+                if left_button == True and right_button == True:
+                    robot.forward()
```

Comment:
Could you comment at least all the GoPiGo3 commands, even if they're totally obvious.